src/java/org/apache/cassandra/io/sstable/format/TOCComponent.java

```diff
@@ -78,20 +80,36 @@ public static Set<Component> loadTOC(Descriptor descriptor, boolean skipMissing)
     }
 
     /**
-     * Appends new component names to the TOC component.
+     * Updates the TOC file by reading existing component entries, merging them with the given components,
+     * sorting the combined list in lexicographic order for deterministic output.
+     *
+     * @param descriptor the SSTable descriptor for which to update the TOC
+     * @param components new components to merge into the TOC (existing TOC entries are preserved)
+     * @throws FSWriteError if an I/O error occurs when creating or overwriting the TOC file
      */
-    public static void appendTOC(Descriptor descriptor, Collection<Component> components)
+    public static void updateTOC(Descriptor descriptor, Collection<Component> components)
     {
+        if (components.isEmpty())
+            return;
+
         File tocFile = descriptor.fileFor(Components.TOC);
-        try (FileOutputStreamPlus out = tocFile.newOutputStream(APPEND);
-             PrintWriter w = new PrintWriter(out))
+
+        Set<String> componentNames = new TreeSet<>(Collections2.transform(components, Component::name));
+
+        if (tocFile.exists())
         {
-            for (Component component : components)
-                w.println(component.name);
-            w.flush();
-            out.sync();
+            List<String> lines = FileUtils.readLines(tocFile);
+            if (lines.isEmpty())
+                throw new FSReadError(new RuntimeException("Unable to read or file is empty"), tocFile.name());
+
+            componentNames.addAll(lines);
         }
-        catch (IOException e)
+
+        try
+        {
+            FileUtils.write(tocFile, new ArrayList<>(componentNames), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
```

Comment:
the previous logic used `out.sync()` method, should we add StandardOpenOption.SYNC flag to preserve the force logic?